Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabDomUtils.cpp

```diff
@@ -582,6 +582,89 @@ namespace AzToolsFramework
                         "PrefabDomUtils::AddNestedInstance - Nested instance already exists in prefab DOM.");
                 }
             }
+
+            bool CheckEntitiesParents(PrefabDom& templateDomRef)
+            {
+                bool result = true;
+                // Search for a TransformComponent and check the "Parent Entity" statement
+                auto sourceIt = templateDomRef.FindMember(PrefabDomUtils::SourceName);
+                if (sourceIt != templateDomRef.MemberEnd() && sourceIt->value.IsString())
```

Comment:
that's a lot of nesting, I would prefer some early returns and early continue so that its easier to read

Reply:
Also can likely be easier to reason about if you had a findInvalidParents or something similar, and then do the removal logic afterwards

Reply:
Made code flatter, using early `return`s and `continue`s.
This inspired adding error logs for possible structural inconsistencies raising `return`s and `continue`s.